events/events_test.go

```diff
@@ -245,3 +251,21 @@ func TestRequestValidate(t *testing.T) {
 		t.Errorf("expected error with invalid subset sample")
 	}
 }
+
+func TestClose(t *testing.T) {
+	t.Parallel()
+	fc := newFakeClient(t, "all", "")
+	resp, err := events.Fetch(fc, events.StartFirst, 0, nil, &events.Filter{Types: []events.Type{""}})
+	if err != nil {
+		t.Fatalf("Received error fetching: %v", err)
+	}
+
+	// Close should be safe to call all the time
+	done := make(chan bool)
+	go func() {
+		resp.Close()
+		close(done)
+	}()
+	resp.Close()
+	<-done
```

Comment:
Would this benefit from being in a switch statement with a time that gives it, say, 5-10 seconds to be done? Then if the timer finished before done, you could fail the test with some sort of output like, "Should have closed by now." That way if it gets broken, it doesn't just hang forever.